### DISCUSSION TITLE
JS - Objects - Constructor Functions - Practice Problems

### DISCUSSION THREAD

**Discussion by akash-coded** [akash-coded/mern#90] [Tasks]
1. Create a constructor function Calculator that creates objects with 3 methods:
    - read() asks for two values using prompt and remembers them in object properties.
    - sum() returns the sum of these properties.
    - mul() returns the multiplication product of these properties.

2. Create a constructor function Accumulator(startingValue).

    Object that it creates should:

    - Store the “current value” in the property value. The starting value is set to the argument of the constructor startingValue.
    - The read() method should use prompt to read a new number and add it to value.

    In other words, the value property is the sum of all user-entered values with the initial value startingValue.

    Here’s the demo of the code:

    ```javascript
    let accumulator = new Accumulator(1); // initial value 1

    accumulator.read(); // adds the user-entered value
    accumulator.read(); // adds the user-entered value

    alert(accumulator.value); // shows the sum of these values
    ```

**Comment by Neethumar** [comment]
### Qn1
```javascript

function Calculator() {
    this.read = function () {
        this.num1 = parseInt(prompt("Enter the first number"));
        this.num2 = parseInt(prompt("Enter the second number"));
    }
    this.sum = function () {
        return `sum is ${this.num1 + this.num2}`;
    }
    this.mul = function () {
        return `Product is ${this.num1 * this.num2}`;
    }
}

const result1 = new Calculator();
result1.read();
console.log(result1.sum());
console.log(result1.mul());

```

### Screenshots
<img width="337" alt="image" src="https://user-images.githubusercontent.com/93718958/183249475-e9f180b2-3f2b-4eec-92ec-d010cbaaa8aa.png">
<img width="335" alt="image" src="https://user-images.githubusercontent.com/93718958/183249496-01f791bd-ee89-4272-87ce-ac66496f821b.png">
<img width="508" alt="image" src="https://user-images.githubusercontent.com/93718958/183249671-7bd3a3cc-e183-4a1f-b64c-caa819118787.png">


### Qn2
```javascript
function Accumulator(startingValue){
    this.value=startingValue
    this.read=function(){
        return this.value+=parseInt(prompt("Enter the value"));
    }

}
let accumulator = new Accumulator(1); // initial value 1

accumulator.read(); // adds the user-entered value
accumulator.read(); // adds the user-entered value

alert(accumulator.value);
```
### Screenshots
<img width="337" alt="image" src="https://user-images.githubusercontent.com/93718958/183249539-489c8f82-cc1b-4045-8c38-49a4fb89609d.png">
<img width="335" alt="image" src="https://user-images.githubusercontent.com/93718958/183249546-f6ee98c8-cad9-4e7f-96a1-2a7bd01dc90b.png">
<img width="335" alt="image" src="https://user-images.githubusercontent.com/93718958/183249564-806af33c-2fa9-458a-8be0-07a54654ad20.png">

**Comment by Nimje1998** [comment]
<!doctype html>
<script src="https://en.js.cx/test/libs.js"></script>
<script src="test.js"></script>
<script>

  function Calculator() {
  
    this.read = function() {
      this.a = +prompt('a?', 0);
      this.b = +prompt('b?', 0);
    };
  
    this.sum = function() {
      return this.a + this.b;
    };
  
    this.mul = function() {
      return this.a * this.b;
    };
  }

</script>

</html>

**OUTPUT:** 
the read method asks for two values using a prompt and remembers them in object properties
when 2 and 3 are entered, the sum is 5
when 2 and 3 are entered, the product is 6

**Comment by Nimje1998** [comment]
<!doctype html>
<script src="https://en.js.cx/test/libs.js"></script>
<script src="test.js"></script>
<script>

  function Accumulator(startingValue) {
    this.value = startingValue;
  
    this.read = function() {
      this.value += +prompt('How much to add?', 0);
    };
  
  }

</script>

</html>

the initial value is the argument of the constructor
after reading 0, the value is 1
after reading 1, the value is 2

**Comment by Amrita-12345** [comment]
## Q.1. Code
```js
function Calculator()
{
   this.read=function()
   {
this.num1=+prompt("Enter the First number");
this.num2=+prompt("Enter the second number");
   };
   this.sum=function()
   {
    return this.num1 + this.num2;
   };
   this.mul=function()
   {
    return this.num1*this.num2;
   };
}
let calculate = new Calculator();
calculate.read();
alert( "Sum of given numbers" +" "+ calculate.sum() );
alert( "Multiplication of given numbers" +" "+ calculate.mul() );
```
## Screenshot
![image](https://user-images.githubusercontent.com/104146616/183249807-ff0ea224-9dc0-49d9-abb0-334e48151ffc.png)
![image](https://user-images.githubusercontent.com/104146616/183249835-8794581e-a777-406e-a70b-106fc7690b92.png)
![image](https://user-images.githubusercontent.com/104146616/183249845-29b964e3-3478-44e3-9919-9afa29834e28.png)
![image](https://user-images.githubusercontent.com/104146616/183249853-68c054c6-9e43-4e8d-a315-2699385d23d7.png)

## Q.2. Code
```js
function Accumulator(startingValue) {
    this.value = startingValue;
  
    this.read = function() {
      this.value += +prompt("Enter the new number");
    };
  }
  let accumulator = new Accumulator(1);
  accumulator.read();
  accumulator.read();
  alert(accumulator.value);
```
## Screenshots
![image](https://user-images.githubusercontent.com/104146616/183250009-5879b4dc-825e-4f29-a228-c35fdb691abb.png)
![image](https://user-images.githubusercontent.com/104146616/183250016-60d76db5-7e22-472a-b8b6-f84a6312e385.png)
![image](https://user-images.githubusercontent.com/104146616/183250054-675559a9-6016-40c2-b324-7044a08af6e0.png)

**Comment by Nikhi54** [comment]
### Que1

```javascript
function Accumulator() {

    this.read = function() {
      this.a = parseInt(prompt(`Enter a first value:`));
      this.b = parseInt(prompt(`Enter a second value:`));
    };
  
    this.sum = function() {
      return this.a + this.b;
    };
  
    this.mul = function() {
      return this.a * this.b;
    };
  }
  
  let accumulator = new Accumulator();
  accumulator.read();
  
  alert( "Sum=" + accumulator.sum() );
  alert( "Mul=" + accumulator.mul() );
```

**Comment by Divyam-09** [comment]
## Code
```Javascript
function Calculator(){
  

   this.read = function(){
    this.num1 = parseInt(prompt("enter num1"));
    this.num2 = parseInt(prompt("enter num2"));
   }

   this.addi = function(){
    
    return this.num1 + this.num2;
   }

   this.mul = function(){
    return this.num1 * this.num2;;
   }
}
let calc = new Calculator();
calc.read()
console.log("the sum of result is" +" "+calc.addi());
console.log("the multiplication result is" +" "+calc.mul());
```

## Screenshot
![image](https://user-images.githubusercontent.com/103888460/183254776-e732b313-a404-4a50-97cc-de5c365e6433.png)
![image](https://user-images.githubusercontent.com/103888460/183254800-12ad203a-5d5b-42f0-871b-1301540e9fd3.png)
![image](https://user-images.githubusercontent.com/103888460/183254810-a150afda-3fc9-4788-87ce-ccfeefaae65d.png)

**Comment by KNiranjani** [comment]
### Q1.
### Code:
```js

function Calculator() {
    this.read = function () {
        this.num1 = parseInt(prompt("Enter the 1st numb"));
        this.num2 = parseInt(prompt("Enter the 2nd numb"));
    }
    this.sum = function () {
        return this.num1 + this.num2;
    }
    this.multiply = function () {
        return this.num1 * this.num2;
    }
}

var compute = new Calculator();
compute.read();
alert("Sum is" +" "+ compute.sum());
alert("Multiplication is" +" "+ compute.multiply());
```
**Screenshot**
![image](https://user-images.githubusercontent.com/37462777/183257585-9c8b1606-f56d-4831-af07-9d441c4b27fb.png)
![image](https://user-images.githubusercontent.com/37462777/183257618-5ee89fd2-5bc9-4cd6-ba5d-470be8919cc0.png)
![image](https://user-images.githubusercontent.com/37462777/183257630-82e237b3-38d3-46d9-9997-75ab70c39f64.png)
![image](https://user-images.githubusercontent.com/37462777/183257649-c158814a-861d-418c-8d95-59e772e71f18.png)


### Q2.
### Code
```js
function Accumulator(startingValue){
    this.value=startingValue
    this.read=function(){
        return this.value+=parseInt(prompt("Enter the value"));
    }

}
let accumulator = new Accumulator(1); // initial value 1

accumulator.read(); // adds the user-entered value
accumulator.read(); // adds the user-entered value

alert(accumulator.value);
```

**Screenshot**
![image](https://user-images.githubusercontent.com/37462777/183256591-e872e230-46e1-47b4-875a-41d3e97c66d9.png)

![image](https://user-images.githubusercontent.com/37462777/183256844-bb4859cf-0020-460d-93eb-511dea1f2d60.png)
![image](https://user-images.githubusercontent.com/37462777/183256866-b5f405ce-d49f-4d40-8d04-6269f810496d.png)

**Comment by shubhamk-coder** [comment]
## Question-1
### Code
```javascript
function Calculator()
{
   this.read=function()
   {
this.num1=+prompt("Enter the First number");
this.num2=+prompt("Enter the second number");
   };
   this.sum=function()
   {
    return this.num1 + this.num2;
   };
   this.mul=function()
   {
    return this.num1*this.num2;
   };
}
let calculate = new Calculator();
calculate.read();
alert( "Sum is " + calculate.sum() + " & Multiplication is " + calculate.mul() );
```
### Screenshot
![image](https://user-images.githubusercontent.com/36131539/184501844-38fdffd0-9c17-4d9d-9853-ff16cee27514.png)
![image](https://user-images.githubusercontent.com/36131539/184501852-fd47236c-be8b-4e84-b602-0507a1b5aa7c.png)
![image](https://user-images.githubusercontent.com/36131539/184501977-0e25d5a6-861b-4e00-875b-8eab0c879f4a.png)
## Question-2
### Code
```javascript
function Accumulator(startingValue) {
  this.value = startingValue;

  this.read = function() {
    this.value += +prompt("Enter the new number");
  };
}
let accumulator = new Accumulator(1);
accumulator.read();
accumulator.read();
alert(accumulator.value);
```
### Screenshot
![image](https://user-images.githubusercontent.com/36131539/184502092-3c1e3974-5074-48f7-a332-ecda8465b4f4.png)
![image](https://user-images.githubusercontent.com/36131539/184502095-0ae0603c-a4ae-40ae-96a7-1bcada861997.png)
![image](https://user-images.githubusercontent.com/36131539/184502114-d4e565e1-a186-4d37-a345-8981777769d0.png)

**Comment by lingamurthi** [comment]
```html
<script>
    
function Calculator() {
  
    this.read()= function() {
         this.a= parseInt(prompt("Enter the first number"))
         this.b= parseInt(prompt("Enter the second number"))
     }

     this.sum()= function() {   
        return this.a+this.b; 
     }
    this.mul()=function() {
        return this.a*this.b; 
    }
}

  </script>



<script>


function Accumulator(startingValue){
    this.value = startingValue;


    this.read=function(){
        return this.value+= parseInt(prompt("Enter new value"));
    }

}
    let accumulator = new Accumulator(1); // initial value 1

accumulator.read(); // adds the user-entered value
accumulator.read(); // adds the user-entered value

alert(accumulator.value); // shows the sum of these values

  </script>


```

**Comment by AbhishekBilagi** [comment]
```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>
<body>
    <script>
        //1
        function calculator(){
            this.read=function(){
                this.a=parseInt(prompt("Enter First Number"));
                this.b=parseInt(prompt("Enter Second Number"));
            }
            this.sum=()=>this.a+this.b;

            this.mul=()=>this.a*this.b;
        }
        let cal=new calculator();
        cal.read();
        document.write("<h1>"+"Sum :"+cal.sum()+"</h1>");
        document.write("<h1>"+"Mul :"+cal.mul()+"</h1>");
       
        //2
        function Accumulator(num){

            this.read=()=>{
                this.c=parseInt(prompt("Enter Number"));
                return this.c+num;
            }
        }
        let accum=new Accumulator(10);
        document.write("<h1>"+"Sum :"+accum.read()+"</h1>");
    </script>
</body>
</html>``
```

**Comment by PurnaPichuka** [comment]
```html
<!-- QUESTION-1 -->
   <script>
        function Calculator()
        {
            let num1 = parseInt(prompt("Enter the First number"))
            let num2 = parseInt(prompt("Enter the second number"))
            this.read=function()
            {
                this.num1 = num1
                this.num2 = num2
                alert(`num1 is ${num1} and num2 is ${num2}`)
            }
            this.sum = function()
            {
                return this.num1 + this.num2
            }
            this.mul=function()
            {
                return this.num1 * this.num2
            }
        }
        let calculate = new Calculator()
        calculate.read();
        let addition = calculate.sum()
        let product = calculate.mul()
        alert(`sum is ${addition} and multiplication is ${product}`)

    </script>

    <!-- QUESTION-2 -->
    <script>
        function Accumulator(startingValue){
            this.value=startingValue
            this.read=function(){
                return this.value += parseInt(prompt("Enter the value"));
            }

        }
        let accumulator = new Accumulator(1); // initial value 1
        accumulator.read(); // adds the user-entered value
        accumulator.read(); // adds the user-entered value

        alert(accumulator.value);
    </script>
```

**Comment by rrajbanshi** [comment]
# 1

```js
function Calculator(){
    this.read = function(){
        this.num1 = +prompt("Enter num1:")
        this.num2 = +prompt("Enter num2:")
    }

    this.sum = function(){
        return this.num1 + this.num2
    }

    this.mul = function(){
        return this.num1 * this.num2
    }
}

let calc = new Calculator();
calc.read()
alert(`sum is ${calc.sum()}`)
alert(`mult is ${calc.mul()}`)

```
# 2
```js

function Accumulator(startingValue){
    this.value = startingValue

    this.read = function(){
        this.value += +prompt("Enter the value:")
    }
}

let accumulator = new Accumulator(1);
accumulator.read()
accumulator.read()
alert(accumulator.value)

```
